python/tvm/runtime/ndarray.py

```diff
@@ -249,6 +249,28 @@ def copyto(self, target):
             return self._copyto(res)
         raise ValueError("Unsupported target type %s" % str(type(target)))
 
+    def create_view(self, shape):
```

Comment:
Let us make it _create_view for now. 

Using views extensively can be unsafe, and should be discouraged. Note that most of TVM function compiled assumes no-alias, which works well when views are not created in frontend.

The particular application of logical/physical layout is a good one that can use the low-level API. but would also be good to think how to make the API safe in the first place(e.g. when specify the physical shape we "consume" the original tensor so it will no longer be used).

Mark it as private with note would help to make this issue clear.

Reply:
Agreed, and updated the name to `_create_view` and the docstring to include a warning.

In the future, I want to make it easier to interact with the different shapes (logical shape, transformed shape, physical shape), but am still working through the different interaction points needed.  With respect to a safe API here, I *think* that would either be `tvm.nd.empty` allow the tensor shape and the shape of the backing allocation to be independently specified, but I haven't finished fleshing out that design.